File: Aleph1.Skeletons.WebAPI/Aleph1.Skeletons.WebAPI.Models/Security/Roles.cs
using System;

namespace Aleph1.Skeletons.WebAPI.Models.Security
{
	/// <summary>User roles</summary>
	[Flags]
	public enum Roles
	{
		/// <summary>Not signed in</summary>
		None = 0,

		/// <summary>Regular user</summary>
		User = 1 << 0,

		/// <summary>Some other role</summary>
		SomeOtherRole = 1 << 1,

		/// <summary>Includes all of the above roles</summary>
		Admin = None | User | SomeOtherRole
	}
}

File: Aleph1.Skeletons.WebAPI/Aleph1.Skeletons.WebAPI.Pages/src/resources/models/roles.ts
/** Authorization roles */
export enum Roles
{
	/** Not logged in */
	None = 0,

	/** Regular user */
	User = 1 << 0,

	/** Another role for demonstration  */
	SomeOtherRole = 1 << 1,

	/** Administrator - includes all below rules */
	Admin = None | User | SomeOtherRole
}
